plugins/wporg-5ftf/includes/stats.php

```diff
@@ -0,0 +1,185 @@
+<?php
+
+/**
+ * Track and display key metrics for the program, to measure growth and effectiveness.
+ */
+
+namespace WordPressDotOrg\FiveForTheFuture\Stats;
+
+use WordPressDotOrg\FiveForTheFuture;
+use WordPressDotOrg\FiveForTheFuture\{ Contributor, Pledge, XProfile };
+use WP_Query;
+
+defined( 'WPINC' ) || die();
+
+add_action( 'init',                 __NAMESPACE__ . '\register_post_types' );
+add_action( 'init',                 __NAMESPACE__ . '\schedule_cron_jobs' );
+add_action( '5ftf_record_snapshot', __NAMESPACE__ . '\record_snapshot' );
+
+add_shortcode( '5ftf_stats', __NAMESPACE__ . '\render_shortcode' );
+
+
+/**
+ * Register the snapshots post type. Each post represents a snapshot of the stats at a point in time.
+ */
+function register_post_types() {
+	$args = array(
+		'supports'     => array( 'custom-fields' ),
+		'public'       => false,
+		'show_in_rest' => true,
+
+		// Only allow posts to be created programmatically.
+		'capability_type' => '5ftf_stats',
+		'capabilities'    => array(
+			'create_posts' => 'do_not_allow',
+		),
+	);
+
+	register_post_type( '5ftf_stats_snapshot', $args );
+}
+
+/**
+ * Schedule the cron job to record a stats snapshot.
+ */
+function schedule_cron_jobs() {
+	if ( wp_next_scheduled( '5ftf_record_snapshot' ) ) {
+		return;
+	}
+
+	// Schedule a repeating "single" event to avoid having to create a custom schedule.
+	wp_schedule_single_event(
+		time() + ( 2 * WEEK_IN_SECONDS ),
+		'5ftf_record_snapshot'
+	);
+}
+
+/**
+ * Record a snapshot of the current stats, so we can track trends over time.
+ */
+function record_snapshot() {
+	$stats = get_snapshot_data();
+
+	$post_id = wp_insert_post( array(
+		'post_type'   => '5ftf_stats_snapshot',
+		'post_author' => 0,
+		'post_title'  => sprintf( '5ftF Stats Snapshot %s', date( 'Y-m-d' ) ),
```

Comment:
Minor, but I don't think the post title needs the `5ftf` prefix.

Reply:
I disagree, but I don't think it's worth the time and energy to debate these kinds of things. In the past I've been very nitpicky in reviews, but now I feel like I was being overzealous, wasting time, and creating unnecessary disagreements. Lately I've only been focusing on things that have a significant impact, like bugs, UX, etc. I think that's made those reviews more productive and less frustrating for everyone.

We're never going to have exactly the same preferences, and trying to ensure 100% uniformity is  exhausting. It also places bureaucratic barriers to devs experimenting with alternative approaches, and evolving their practices. We have a long list of impactful projects in the backlog, and I'd rather spend my limited time and energy shipping those than re-writing and re-testing things that already work well. 

If the code works as expected, is easy to understand, doesn't introduce technical debt, etc, then I think minor quibbles like this should just be ignored. If you feel strongly, you're always welcome to change it after it's merged.